comiccrawler/grabber.py

```diff
@@ -20,6 +22,30 @@
 	"Accept-Encoding": "gzip, deflate"
 }
 
+cooldown = {}
+grabber_pool = {}
+grabber_pool_lock = Lock()
+
+@contextmanager
+def get_request_lock(url):
+	domain = urlsplit(url).hostname
+	defer = Defer()
+	try:
+		with grabber_pool_lock:
+			last_defer = grabber_pool.get(domain)
+			grabber_pool[domain] = defer
+		if last_defer:
+			last_defer.get()
+		yield
+	except Exception as err:
+		defer.reject(err)
+		raise
+	finally:
+		@async_
+		def _():
+			time.sleep(cooldown.get(domain, 0))
```

Comment:
@DummySYSTEM You may want to change this line into
```python
delay = cooldown.get(domain, 0)
if delay != 0:
  delay += random.uniform(-0.5, 0.5)
time.sleep(delay)
```

Reply:
get your point now